src/graphql/resolvers.ts

```diff
@@ -52,9 +59,95 @@ export const loaders: MercuriusLoaders = {
           },
         },
       });
-      const metadataBySubject = keyBy(metadata, "subject");
+      const metadataBySubject = _.keyBy(metadata, "subject");
       return queries.map(({ obj: asset }) => metadataBySubject[asset.id] || null);
     },
+    async cip68ft(queries, ctx) {
+      const cip68assets = queries.filter(({ obj: asset }) => asset.assetName.startsWith(ASSET_LABELS.FT));
+
+      if (cip68assets.length === 0) {
+        return queries.map(() => null);
+      }
+
+      const referenceMap = new Map(
+        cip68assets.map(({ obj: asset }) => [
+          asset.id,
+          `${asset.policyId}${getRefereneAssetName(asset.assetName)}`,
+        ])
+      );
+
+      const metadata = await ctx.db.CIP68Metadata.findAll({
+        where: {
+          subject: {
+            [Op.in]: _.compact(cip68assets.map(({ obj: asset }) => referenceMap.get(asset.id))),
+          },
+        },
+      });
+
+      const metadataBySubject = _.keyBy(metadata, "subject");
+      return queries.map(({ obj: asset }) => {
+        const referenceSubject = referenceMap.get(asset.id);
+        if (!referenceSubject) {
+          return null;
+        }
+        const metadata = metadataBySubject[referenceSubject];
+        if (!metadata) {
+          return null;
+        }
+        const extraMetadata = safeJSON.parse(metadata.otherProperties || "{}");
+        return {
+          subject: asset.id,
+          name: metadata.name,
+          description: metadata.description || "",
+          policy: null,
+          ticker: extraMetadata.ticker || null,
+          url: extraMetadata.url || null,
+          logo: extraMetadata.logo || null,
+          decimals: extraMetadata.decimals || null,
+        };
+      });
+    },
+    async cip68nft(queries, ctx) {
+      const cip68assets = queries.filter(({ obj: asset }) => asset.assetName.startsWith(ASSET_LABELS.NFT));
+
+      if (cip68assets.length === 0) {
+        return queries.map(() => null);
+      }
+
+      const referenceMap = new Map(
+        cip68assets.map(({ obj: asset }) => [
+          asset.id,
+          `${asset.policyId}${getRefereneAssetName(asset.assetName)}`,
+        ])
+      );
+
+      const metadata = await ctx.db.CIP68Metadata.findAll({
+        where: {
+          subject: {
+            [Op.in]: _.compact(cip68assets.map(({ obj: asset }) => referenceMap.get(asset.id))),
+          },
+        },
+      });
+      const metadataBySubject = _.keyBy(metadata, "subject");
+      return queries.map(({ obj: asset }) => {
+        const referenceSubject = referenceMap.get(asset.id);
+        if (!referenceSubject) {
+          return null;
+        }
```

Comment:
quite a bit of code duplication. If the cip68 draft doesn't get changed, clean this up.